src/solver/expressions/visitors/CompareVisitor.cpp

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2007-2024, RTE (https://www.rte-france.com)
+ * See AUTHORS.txt
+ * SPDX-License-Identifier: MPL-2.0
+ * This file is part of Antares-Simulator,
+ * Adequacy and Performance assessment for interconnected energy networks.
+ *
+ * Antares_Simulator is free software: you can redistribute it and/or modify
+ * it under the terms of the Mozilla Public Licence 2.0 as published by
+ * the Mozilla Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * Antares_Simulator is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Mozilla Public Licence 2.0 for more details.
+ *
+ * You should have received a copy of the Mozilla Public Licence 2.0
+ * along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+ */
+#include <antares/solver/expressions/nodes/ExpressionsNodes.h>
+#include <antares/solver/expressions/visitors/CompareVisitor.h>
+
+template<class T, class V>
+static bool compareBinaryNode(V& visitor, const T& node, const Antares::Solver::Nodes::Node& other)
```

Comment:
And let's be crazy and try to remove the template parameter : 
Could we have something like this : 
```C++
bool CompareVisitor::operator==(const Nodes::BinaryNode* node1, 
                                const Nodes::Node* node2)
{
    if (typeid(node1) == typeid(node2))
    {
        bool left = dispatch(*node1.left(), *node2->left());
        bool right = dispatch(*node1.right(), *node2->right());
        return left && right;
    }
    return false;
}
```
?